File: airbyte-integrations/connectors/destination-e2e-test/src/main/java/io/airbyte/integrations/destination/e2e_test/logging/TestingLogger.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.e2e_test.logging;

import io.airbyte.protocol.models.v0.AirbyteRecordMessage;

public interface TestingLogger {

  enum LoggingType {
    FirstN,
    EveryNth,
    RandomSampling
  }

  void log(AirbyteRecordMessage recordMessage);

}

File: airbyte-integrations/connectors/destination-e2e-test/src/main/kotlin/io/airbyte/integrations/destination/e2e_test/logging/TestingLogger.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.integrations.destination.e2e_test.logging

import io.airbyte.protocol.models.v0.AirbyteRecordMessage

interface TestingLogger {
    enum class LoggingType {
        FirstN,
        EveryNth,
        RandomSampling
    }

    fun log(recordMessage: AirbyteRecordMessage?)
}
